Feat: enhance code with annotations

Djtheman1 opened the pull request.

Commit:
feat: add annotation in AvatarContainer.tsx

Commit:
feat: annotation in AavatrScene.tsx

Djtheman1 committed:
Feat: Adds annotated code summary for both developers and none developers

Djtheman1 committed:
Merge branch 'CodeAnnotation' of https://github.com/code-differently/UD-Athletics-App into CodeAnnotation

Commit:
fix: specific annotation in AvatarContainer

Commit:
fix: annotate AvatarScene

descriptions included

Commit:
rm: header to polish up

anthonydmays commented on src/app/layout.tsx:
```diff
@@ -5,23 +5,29 @@ import type { Metadata } from "next";
 
 const inter = Inter({ subsets: ["latin"] });
 
+// Define metadata for the page, including the title and description
```
Before I review this, I want you to carefully read the following document and adjust or remove comments accordingly. Tl;dr: Focus on the "why", not the "what," and only when the code doesn't already make it clear (assuming someone has already been taught what you have been taught and got a good grade).

https://danielrotter.at/2021/01/16/code-comments-are-mostly-a-violation-of-dry.html#:~:text=%E2%80%9CDon't%20repeat%20yourself%E2%80%9D,its%20own%20acronym%20(DRY).

You should probably need to delete at least 50% of the comments you've added as a result. 👍🏾

anthonydmays commented on src/app/layout.tsx:
```diff
@@ -5,23 +5,29 @@ import type { Metadata } from "next";
 };
 
+// RootLayout component wraps all pages, ensuring a consistent structure
 export default function RootLayout({
```
Use block comments for public (exported) functions like this:

```Typescript
/**
 * The root layout for all pages.
 */
```

anthonydmays commented on src/app/layout.tsx:
```diff
@@ -5,23 +5,29 @@ import type { Metadata } from "next";
       <head>
+        {/* Meta tag to ensure responsive design on different screen sizes */}
         <meta name="viewport" content="width=device-width, initial-scale=1" />
         {/* Add more meta tags here if needed */}
```
Definitely remove comments like this.

AmiyahJo replied:
All comments with the '{/* same format ? */}

anthonydmays commented on src/components/HelpIcon.tsx:
```diff
@@ -1,16 +1,39 @@
+// When you click the button, it triggers a specific action (like opening a help menu or guide). 
+// The button is carefully styled to look good and is positioned at a specific spot on the screen. 
+// Essentially, it’s a neat and tidy way to add a help feature to a webpage.
+
```
Only summary for developers, using the block comment style.

anthonydmays commented on src/components/HelpIcon.tsx:
```diff
@@ -1,16 +1,39 @@
+// Import the HelpCircle icon component from the lucide-react library
```
Don't comment imports.

Commit:
fix: layout.tsx comments

anthonydmays commented on src/components/avatar/AvatarScene.tsx:
```diff
@@ -1,86 +1,109 @@
-    const light = new THREE.AmbientLight(0xffffff, 0.5);
+    if (!containerRef.current) return; // Stop if there's no container to place the 3D model in.
+    
+    // **Setting up the 3D scene**
```
No asterisks, just `// Setting up the 3D scene`.

anthonydmays commented on src/components/avatar/AvatarScene.tsx:
```diff
@@ -1,86 +1,109 @@
+    
+    // **Setting up the 3D scene**
+    const scene = new THREE.Scene(); // Creates an empty 3D world.
+    scene.background = new THREE.Color(0xf0f0f0); //background color of the scene (currently light gray)
```
The value `0xf0f0f0` is a magic number. Store in a CONSTANT named for the actual color and use here, like so:

```Typescript
const LIGHT_GRAY_HEX = 0xf0f0f0;

const AvatarScene = () => {
   ...
   scene.background = new THREE.Color(LIGHT_GRAY_HEX);
}
``` 

Please replace all the magic numbers throughout this file with properly defined constants.

Commit:
fix: AvatarScene.tsx comments and magic numbers

removed import comments , magic numbers , added constants , and changed some to block format

Commit:
fix: avatarScene constant comment changed to block comments

Commit:
rm: AvatarContainer comments

they simply restate the code included

Commit:
rm: comments in AvatarScene
fix: replace magic numbers with constants

Djtheman1 committed:
Feat: Adds DRY principle to code.

Djtheman1 committed:
Merge branch 'CodeAnnotation' of https://github.com/code-differently/UD-Athletics-App into CodeAnnotation

Commit:
fix: avatarScene component comments , and layout.tsx

Commit:
rm: duplicate comments on self-explanatory code

Commit:
rm: header import
